docs/software_requirements/timers.sdoc

```diff
@@ -26,12 +23,334 @@ UID: ZEP-SRS-4-2
+TITLE: Initialialization with stop function
+STATEMENT: >>>
+When initializing a timer, the stop function a timer can have shall be set.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-10
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer start
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to start a defined and initialized timer.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-11
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Start timer status reset
+STATEMENT: >>>
+When a timer is started with the start mechanism the status of the timer is set to zero.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-12
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Start timer counting values
+STATEMENT: >>>
+When a timer is started with the start mechanism, the timer starts counting with the given persiod and duration values
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-13
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer stop
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to stop a running timer prematurely.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-14
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer stop invocation
+STATEMENT: >>>
+When a Timer is stopped with the provide stop mechanism, the stop function is called if one is defined.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-15
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer status read
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to read the status of a timer.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-16
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer status read reset
+STATEMENT: >>>
+When the timer is read via the timer status read mechanism, the status shall be reset.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-17
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer status read reset
+STATEMENT: >>>
+When the timer is read via the timer status read mechanism, the status shall be reset.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-18
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer thread synchronization
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to synchronize a thread with a defined and initialized timer.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-19
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer expire when thread synchronization is used
+STATEMENT: >>>
+When the thread synchronization mechanism is used, the thread shall be blocked until the timer expires.
```

Comment:
I think that more than one thread can synchronize with a given timer. It might have a side-effect if another thread want's to read the "status" as well and will take some action towards that "status" but isn't that more like a risk of the application implementation itself how this can be used`?

What do you mean exactly with safe operation?